addons/l10n_jo_edi/models/account_move.py

```diff
@@ -0,0 +1,258 @@
+        string="Jordan E-Invoice XML",
+        compute=lambda self: self._compute_linked_attachment_id(
+            "l10n_jo_edi_xml_attachment_id", "l10n_jo_edi_xml_attachment_file"
+        ),
+        depends=["l10n_jo_edi_xml_attachment_file"],
+        help="Jordan: the most recent e-invoice XML.",
+    )
+
+    def _get_jo_invoice_type_number(self):
+        self.ensure_one()
+        return INVOICE_NUMBERS[self.company_id.l10n_jo_edi_taxpayer_type][self.move_type]
+
+    def _get_payment_method(self):
+        self.ensure_one()
+        return PAYMENT_CODES_MAP[self.company_id.l10n_jo_edi_taxpayer_type]['receivable']
+
+    def _get_type_code(self):
+        self.ensure_one()
+        if self.move_type == 'out_invoice':
+            return "388"
+        elif self.move_type == 'out_refund':
+            return "381"
+
+    def _aggregate_lines_jod(self, aggregation_fn_name):
+        self.ensure_one()
+        total = 0
+        for line in self.invoice_line_ids:
+            total += getattr(line, aggregation_fn_name)()
+        return total
+
+    def _get_discount_amount_jod(self):
+        return self._aggregate_lines_jod('_get_discount_amount_jod')
+
+    def _get_tax_exclusive_amount_jod(self):
+        return self._aggregate_lines_jod('_get_tax_exclusive_amount_jod')
+
+    def _get_tax_inclusive_amount_jod(self):
+        return self._aggregate_lines_jod('_get_tax_inclusive_amount_jod')
+
+    @api.depends("country_code", "move_type")
+    def _compute_l10n_jo_edi_is_needed(self):
+        for move in self:
+            move.l10n_jo_edi_is_needed = (
+                move.country_code == "JO"
+                and move.move_type in ("out_invoice", "out_refund")
+            )
+
+    @api.depends("l10n_jo_edi_is_needed", "state", "l10n_jo_edi_xml_attachment_id", "l10n_jo_edi_error", "reversal_move_id", "payment_state")
+    def _compute_jo_edi_state(self):
+        jo_moves = self.filtered(lambda move: move.l10n_jo_edi_is_needed)
+        for move in jo_moves:
+            if move.state == 'posted':
+                move.l10n_jo_edi_state = 'to_send'
+            if move.l10n_jo_edi_xml_attachment_id and not move.l10n_jo_edi_error:
+                move.l10n_jo_edi_state = 'sent'
+            if move.reversal_move_id and move.payment_state != 'partial':
+                move.l10n_jo_edi_state = 'to_cancel'
+            if move.state == 'cancel' or move.payment_state == 'reversed':
+                move.l10n_jo_edi_state = 'cancelled'
+            if move.state == 'draft':
+                move.l10n_jo_edi_state = None
+
+    @api.depends("l10n_jo_edi_xml_attachment_id", "l10n_jo_edi_error")
+    def _compute_show_reset_to_draft_button(self):
+        super()._compute_show_reset_to_draft_button()
+        self.filtered(lambda move: move.l10n_jo_edi_xml_attachment_id and not move.l10n_jo_edi_error).show_reset_to_draft_button = False
+
+    def button_draft(self):
+        # EXTENDS 'account'
+        self.write(
```

Comment:
`l10n_jo_edi_xml_attachment_id` is computed. No need to write an explicit value on it